crates/kubelet/src/config.rs

```diff
@@ -85,77 +122,240 @@ impl Config {
         })
     }
 
+    fn new_from_builder(builder: ConfigBuilder) -> Self {
+        let fallbacks = ConfigBuilderFallbacks {
+            hostname: || default_hostname().expect("unable to get default hostname"),
+            data_dir: || default_data_dir().expect("unable to get default data directory"),
+            cert_path: default_cert_path,
+            key_path: default_key_path,
+            node_ip: |hn, ip| default_node_ip(hn, ip).expect("unable to get default node IP"),
+        };
+        let build_result = ConfigBuilder::build(builder, fallbacks);
+        build_result.unwrap() // TODO: assuming okay to panic since that's what fallbacks do
+    }
+
+    /// Parses the specified config file and sets the proper defaults.
+    /// If the specified file does not exist, this function panics.
+    /// It is up to callers of the function to ensure any file they specify exists.
+    pub fn new_from_file_only(filename: PathBuf) -> Self {
+        let builder = ConfigBuilder::from_config_file(filename).unwrap();
+        Config::new_from_builder(builder)
+    }
+
     /// Parses all command line flags and sets the proper defaults. The version
     /// of your application should be passed to set the proper version for the CLI
     #[cfg(any(feature = "cli", feature = "docs"))]
     #[cfg_attr(feature = "docs", doc(cfg(feature = "cli")))]
-    pub fn new_from_flags(version: &str) -> Self {
-        // TODO: Support config files too. config-rs and clap don't just work
-        // together, so there is no easy way to merge together everything right
-        // now. This function is here so we can do that data massaging and
-        // merging down the road
+    pub fn new_from_flags_only(version: &str) -> Self {
```

Comment:
Nit: I don't think the `only` is needed in the function name here or in `new_from_file`